src/Features/Traits/HttpResponse.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\BLT\Features\Traits;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\CssSelector\CssSelectorConverter;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\HttpFoundation\Response;
+
+trait HttpResponse
+{
+    protected Response $response;
+
+    /**
+     * @When a response is received
+     */
+    public function aResponseIsReceived(): void
+    {
+        $this->response = $this->getContainer()->handle($this->request);
+    }
+
+    /**
+     * @Then a response status code should be :status
+     */
+    public function aResponseStatusCodeShouldBe(int $status): void
+    {
+        Assert::assertEquals($status, $this->response->getStatusCode());
+    }
+
+    /**
+     * @Given a response HTML should contain :phrase phrase
+     */
+    public function aResponseHTMLShouldContainPhrase(string $phrase): void
+    {
+        Assert::assertStringContainsString($phrase, $this->response->getContent());
+    }
+
+    /**
+     * @Given a response HTML should contain CSRF token
+     */
+    public function aResponseHTMLShouldContainCSRFToken(): void
+    {
+        $converter = new CssSelectorConverter();
+        $xPath = $converter->toXPath("form input[name=\"_token\"]");
+
+        $crawler = new Crawler($this->response->getContent());
+        $node = $crawler->filterXPath($xPath)->first();
+
+        Assert::assertNotNull($node?->attr("value"));
+    }
+
+    /**
+     * @Then the response should be JSON
+     */
+    public function aResponseShouldBeJson(): void
+    {
+        Assert::assertJson($this->response->getContent());
+    }
+
+    /**
+     * @Then the response should contain JSON:
+     */
+    public function aResponseShouldContainJson(TableNode $table): void
+    {
+        $json = json_decode($this->response->getContent(), true);
+
+        foreach ($table->getRowsHash() as $key => $value) {
+            Assert::assertArrayHasKey($key, $json);
```

Comment:
This would be nice if we could use dot notation on arrays. Same thing that is used for validation for example.

Please consider creating a new issue for that.

Reply:
For validation probably too.

Reply:
Okey issue created. I will change this file in the "Dot Notation PR" so for now can it stay like that ?